compiler/rustc_middle/src/infer/canonical.rs

```diff
@@ -63,7 +63,7 @@ impl<'tcx> ty::TypeFoldable<'tcx> for CanonicalVarInfos<'tcx> {
 /// variables. You will need to supply it later to instantiate the
 /// canonicalized query response.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, TyDecodable, TyEncodable)]
-#[derive(HashStable, TypeFoldable, TypeVisitable, Lift)]
+#[derive(HashStable, TypeFoldable, TypeVisitable, Lift, Default)]
```

Comment:
that seems dangerous. The `Default` for var values should semantically be the identity substitutions. That cannot be done in a `Default` impl.

I think that having the default be the empty list is error prone as it can easily lead to us not providing any substitutions at all.